drafts/25-xxx-specifications.txt

```diff
@@ -555,16 +555,59 @@ nd20. The result of evaluating a processor-dependent directive is processor-depe
 4 Macro recognition and expansion
 ---------------------------------
 
+This section is currently incomplete.  A self-contained specification
+for macro recognition and expansion will appear in a forthcoming
+paper.
+
+4.1 Comparison to Macro recognition and expansion in CPP
+--------------------------------------------------------
+
+We intend for macro recognition and expansion to operate almost
+exactly as specified in C 2023 section 6.10.5, with the following
+minor exceptions:
+
+me10. When determining argument boundaries in the invocation of a
+      function-like macro, FPP ignores commas surrounded by matching
+      sets of `[  ]` or `(/  /)` brackets, in addition to matching
+      sets of `(  )` parentheses (as specified in C 2023 section
+      6.10.5-11).
+
+me30. As specified by C 2023 6.10.5-10: "Within the sequence of
+      preprocessing tokens making up an invocation of a function-like
+      macro, new-line is considered a normal white-space character."
+
+me31. In addition, within the sequence of preprocessing tokens making
+      up an invocation of a function-like macro, a `&` character
+      immediately followed by a newline is immediately replaced
+      by a single blank character before subsequent processing.
```

Comment:
Similarly, I think me31 is problematic. Consider these two cases

```fortran
#define atan(x,y) ATAN(x,y)
      real :: g = atan(ab&
                          &c,
                          y)
```
and
```fortran
      real :: g = atan(ab&
                          &c,
                           y)
```

The Fortran continuation rules are weird; an `&` at the end of a line and at the beginning of the next line removes the whitespace between the two lines.

So me31 leaves the FLM-like macro in the first case with a dangling `&` on the next line, leaving us with the separate tokens `ATAN` `(` `ab` `&` `c` `,` `y` `)`.

The `atan` function call in the second instance in normal Fortran source becomes a call to `atan(abc, y)`. Tokens are the obvious ones.